backend/src/main/java/org/eclipse/tractusx/puris/backend/supply/logic/service/SupplierSupplyService.java

```diff
@@ -0,0 +1,173 @@
+import org.eclipse.tractusx.puris.backend.stock.logic.dto.itemstocksamm.DirectionCharacteristic;
+import org.eclipse.tractusx.puris.backend.stock.logic.service.MaterialItemStockService;
+import org.eclipse.tractusx.puris.backend.supply.domain.model.OwnSupplierSupply;
+import org.eclipse.tractusx.puris.backend.supply.domain.model.ReportedSupplierSupply;
+import org.eclipse.tractusx.puris.backend.supply.domain.repository.ReportedSupplierSupplyRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SupplierSupplyService {
+    private final ReportedSupplierSupplyRepository repository; 
+    private final PartnerService partnerService;
+    private final OwnDeliveryService ownDeliveryService;
+    private final ReportedDeliveryService reportedDeliveryService;
+    private final OwnProductionService productionService;
+    private final MaterialItemStockService stockService;
+    private final MaterialServiceImpl materialService;
+
+    protected final Function<ReportedSupplierSupply, Boolean> validator;
+
+    public SupplierSupplyService(
+        ReportedSupplierSupplyRepository repository,
+        PartnerService partnerService,
+        OwnDeliveryService ownDeliveryService,
+        ReportedDeliveryService reportedDeliveryService,
+        OwnProductionService productionService,
+        MaterialItemStockService stockService,
+        MaterialServiceImpl materialService) {
+        this.repository = repository;
+        this.partnerService = partnerService;
+        this.ownDeliveryService = ownDeliveryService;
+        this.reportedDeliveryService = reportedDeliveryService;
+        this.productionService = productionService;
+        this.stockService = stockService;
+        this.materialService = materialService;
+        this.validator = this::validate;
+    }
+
+        public final List<OwnSupplierSupply> calculateSupplierDaysOfSupply(String material, String partnerBpnl, String siteBpns, int numberOfDays) {
+        List<OwnSupplierSupply> supplierSupply = new ArrayList<>();
+        LocalDate localDate = LocalDate.now();
+
+        List<Double> ownDeliveries = ownDeliveryService.getQuantityForDays(material, partnerBpnl, siteBpns, DirectionCharacteristic.OUTBOUND, numberOfDays);
+        List<Double> reportedDeliveries = reportedDeliveryService.getQuantityForDays(material, partnerBpnl, siteBpns, DirectionCharacteristic.OUTBOUND, numberOfDays);
+        List<Double> deliveries = mergeDeliveries(ownDeliveries, reportedDeliveries);
+        List<Double> productions = productionService.getQuantityForDays(material, partnerBpnl, siteBpns, numberOfDays);
+        double stockQuantity = stockService.getInitialStockQuantity(material, partnerBpnl);
+        
+        for (int i = 0; i < numberOfDays; i++) {
+            Date date = Date.from(localDate.atStartOfDay(ZoneId.systemDefault()).toInstant());
+
+            if (i == numberOfDays - 1) {
```

Comment:
Why do we only add productions on the last day to the stock to use?

Reply:
We're not, its just for the last iteration, I need to add productions/deliveries before going to the getDaysOfSupply. Maybe I complicated it though..